Source/ForecastDeskbarView.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2017 Benjamin Amos
+ * All rights reserved. Distributed under the terms of the MIT license.
+ */
+
+#include <Bitmap.h>
+#include <StringView.h>
+#include <String.h>
+#include <MessageRunner.h>
+#include <Looper.h>
+#include <Roster.h>
+
+#include "ForecastDeskbarView.h"
+#include "ForecastView.h"
+#include "App.h"
+#include "Util.h"
+
+const uint32 kUpdateForecastMessage = 'Updt';
+const float kToolTipDelay = 1000000; /*1000000ms = 1s*/
+
+ForecastDeskbarView::ForecastDeskbarView(BRect viewSize, ForecastView* forecastView)
+	:	BView(viewSize, "ForecastDeskbarView", B_FOLLOW_ALL, B_WILL_DRAW | B_NAVIGABLE | B_PULSE_NEEDED)
+{
+	fForecastView = forecastView;
+}
+
+ForecastDeskbarView::ForecastDeskbarView(BMessage* archive)
+	:	BView(BRect(0, 0, 15, 15), "ForecastDeskbarView", B_FOLLOW_ALL, B_WILL_DRAW)
+{
+	fForecastView = NULL;
+}
+
+ForecastDeskbarView::~ForecastDeskbarView()
+{
+	delete fMessageRunner;
```

Comment:
fMessageRunner should be initialized to NULL in all constructors, otherwise this will crash if the view is deleted without being attached to a window.

Reply:
This should be done now. See https://github.com/HaikuArchives/Weather/pull/52/commits/db7290bb822682a33e5407484442c5ec7e49496c#diff-e08bdec831dbadaf604238e13a9cfa37R25.